File: TripKitAndroid/src/main/java/com/skedgo/tripkit/TripPreferences.java
package com.skedgo.tripkit;

import com.skedgo.tripkit.data.tsp.RegionInfo;

import io.reactivex.Observable;

public interface TripPreferences {
    /**
     * This option should be used when {@link RegionInfo#supportsConcessionPricing()} is true.
     */
    boolean isConcessionPricingPreferred();

    void setConcessionPricingPreferred(boolean isConcessionPricingPreferred);

    /**
     * @return An {@link Observable} which emits value of {@link #isConcessionPricingPreferred()}
     * when it has changed.
     */
    Observable<Boolean> whenConcessionPricingPreferenceChanges();

    /**
     * This option should be used when {@link RegionInfo#transitWheelchairAccessibility()} is true.
     */
    boolean isWheelchairPreferred();

    void setWheelchairPreferred(boolean isWheelchairPreferred);

    /**
     * @return An {@link Observable} which emits value of {@link #isWheelchairPreferred()}
     * when it has changed.
     */
    Observable<Boolean> whenWheelchairPreferenceChanges();
}

File: TripKitAndroid/src/main/java/com/skedgo/tripkit/TripPreferences.kt
package com.skedgo.tripkit

import io.reactivex.Observable

interface TripPreferences {

    /**
     * This option should be used when [RegionInfo.supportsConcessionPricing] is true.
     */
    fun isConcessionPricingPreferred(): Boolean

    fun setConcessionPricingPreferred(isConcessionPricingPreferred: Boolean)

    /**
     * An [Observable] which emits the value of [isConcessionPricingPreferred] when it changes.
     */
    fun whenConcessionPricingPreferenceChanges(): Observable<Boolean>

    /**
     * This option should be used when [RegionInfo.transitWheelchairAccessibility] is true.
     */
    fun isWheelchairPreferred(): Boolean

    fun setWheelchairPreferred(isWheelchairPreferred: Boolean)

    /**
     * An [Observable] which emits the value of [isWheelchairPreferred] when it changes.
     */
    fun whenWheelchairPreferenceChanges(): Observable<Boolean>
}
